### PR TITLE
Add option to run patchman behind traefik

### PR DESCRIPTION
**Pull request by berendt**
Signed-off-by: Christian Berendt <berendt@osism.tech>

### CODE REVIEW

File: roles/patchman/handlers/main.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+- name: Restart patchmang service
+  become: true
+  ansible.builtin.service:
+    name: "{{ patchmang_service_name }}"
```
Comment (osfrickler) [inline roles/patchman/handlers/main.yml:5]: dito

File: roles/patchman/handlers/main.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+- name: Restart patchmang service
```
Comment (osfrickler) [inline roles/patchman/handlers/main.yml:2]: mange tak for fixing the typo

File: roles/patchman/templates/docker-compose.yml.j2 [hunk 3]
```diff
@@ -48,3 +67,7 @@ networks:
       driver: default
       config:
         - subnet:  {{ patchman_network }}
+{% if nexus_traefik|bool %}
```
Comment (osfrickler) [inline roles/patchman/templates/docker-compose.yml.j2:70]: s/nexus/patchman/

File: roles/patchman/defaults/main.yml [hunk 4]
```diff
@@ -64,3 +64,14 @@ patchman_postgres_databasename: "{{ patchman_postgres_username }}"
 
 patchman_memcached_tag: alpine
 patchman_memcached_image: "{{ docker_registry_memcached }}/library/memcached:{{ patchman_memcached_tag }}"
+
+##########################
+# traefik
+
+patchman_traefik: false
+
+patchman_traefik_path_prefix: patchman
```
Comment (osfrickler) [inline roles/patchman/defaults/main.yml:73]: No longer needed?

File: roles/patchman_client/tasks/update.yml [hunk 5]
```diff
@@ -1,8 +1,11 @@
 ---
 - name: Wait for patchman service
-  ansible.builtin.wait_for:
-    host: "{{ patchman_client_host }}"
-    port: "{{ patchman_client_port }}"
+  ansible.builtin.uri:
+    url: "{{ patchman_client_server_url }}"
```
Comment (osfrickler) [inline roles/patchman_client/tasks/update.yml:4]: Do you also want to change the definition of patchman_client_server_url to be depending on patchman_traefik or do you expect deployers to override it accordingly?

Comment (berendt) [reply roles/patchman_client/tasks/update.yml:4]: Not sure. The deployer should take care of it for the moment.

**Pull request merged**